goose-recipes/triage-issue.yaml

```diff
@@ -0,0 +1,130 @@
+version: 0.1.0
+title: "Triage and analyze RHEL package issues"
+description: "Analyze Jira issues to determine the appropriate course of action: rebase, backport, request clarification, or no action needed."
+
+instructions: |
+  - Never create, delete, update or modify an Issue in Jira
+  - Follow the decision guidelines strictly to determine the correct action
+
+parameters:
+- key: issue
+  input_type: string
+  requirement: required
+  description: Jira issue identifier to analyze (e.g., RHEL-12345)
+
+
+prompt: |
+  You are a seasoned RHEL (Red Hat Enterprise Linux) software engineer. Your expertise is in analyzing component issues and identifying the most efficient path to resolution, whether through a version rebase, a patch backport, or by requesting clarification when blocked.
+
+  Goal: Analyze the given issue to determine the correct course of action.
+
+  ** Initial Analysis Steps**
+
+  1. Open the {{ issue }} Jira issue and thoroughly analyze it:
+     * Extract key details from the title, description, fields, and comments
+     * Pay special attention to comments as they often contain crucial information such as:
+       - Additional context about the problem
+       - Links to upstream fixes or patches
+       - Clarifications from reporters or developers
+       - Related issues or duplicate reports
+     * Look for keywords indicating the root cause of the problem
+     * Identify specific error messages, log snippets, or CVE identifiers
+     * Note any functions, files, or methods mentioned
+     * Pay attention to any direct links to fixes provided in the issue
+
+  2. Identify the package name that must be updated:
+     * Determine the name of the package from the issue details
+     * Confirm the package repository exists by running `git ls-remote https://gitlab.com/redhat/centos-stream/rpms/<package_name>`
+     * A successful command (exit code 0) confirms the package exists
+     * If the package does not exist, re-examine the Jira issue for the correct package name
+
+  3. Identify the target branch for updates:
+     * Look at the fixVersion field in the Jira issue to determine the target branch
+     * Apply the mapping rule: fixVersion named rhel-N maps to branch named cNs
+     * Verify the branch exists on GitLab
+     * This branch information will be needed for both rebases and backports
+
+  4. Proceed to decision making process described below.
+
+  **Decision Guidelines & Investigation Steps**
+
+  You must decide between one of 4 actions. Follow these guidelines to make your decision:
+
+  1. Rebase
+     A Rebase is only to be chosen when the issue explicitly instructs you to "rebase" or "update" to a newer/specific upstream version. Do not infer this.
+
+  2. Backport a Patch OR Request Clarification
```

Comment:
The agent should be able to assign story points based on complexity of the fix. It is encouraged by the product owners to better understand goals of the sprints.

Reply:
considering the automation would handle the followup actions, I am not sure this really makes sense